src/page-modules/contact/formInputValidator.ts

```diff
@@ -0,0 +1,302 @@
+        },
+        {
+          field: 'line',
+          errorMessage: PageText.Contact.inputFields.line.errorMessages.empty,
+        },
+        {
+          field: 'fromStop',
+          errorMessage:
+            PageText.Contact.inputFields.fromStop.errorMessages.empty,
+        },
+        {
+          field: 'toStop',
+          errorMessage: PageText.Contact.inputFields.toStop.errorMessages.empty,
+        },
+        {
+          field: 'reasonForTransportFailure',
+          errorMessage:
+            PageText.Contact.inputFields.reasonForTransportFailure.errorMessages
+              .empty,
+        },
+        {
+          field: 'firstName',
+          errorMessage:
+            PageText.Contact.inputFields.firstName.errorMessages.empty,
+        },
+        {
+          field: 'lastName',
+          errorMessage:
+            PageText.Contact.inputFields.lastName.errorMessages.empty,
+        },
+        {
+          field: 'email',
+          errorMessage: PageText.Contact.inputFields.email.errorMessages.empty,
+        },
+        {
+          field: 'address',
+          errorMessage:
+            PageText.Contact.inputFields.address.errorMessages.empty,
+        },
+        {
+          field: 'postalCode',
+          errorMessage:
+            PageText.Contact.inputFields.postalCode.errorMessages.empty,
+        },
+        {
+          field: 'city',
+          errorMessage: PageText.Contact.inputFields.city.errorMessages.empty,
+        },
+        {
+          field: 'phoneNumber',
+          errorMessage:
+            PageText.Contact.inputFields.phoneNumber.errorMessages.empty,
+        },
+      );
+
+    case 'other':
+      fields.push(
+        {
+          field: 'firstName',
+          errorMessage:
+            PageText.Contact.inputFields.firstName.errorMessages.empty,
+        },
+        {
+          field: 'lastName',
+          errorMessage:
+            PageText.Contact.inputFields.lastName.errorMessages.empty,
+        },
+        {
+          field: 'email',
+          errorMessage: PageText.Contact.inputFields.email.errorMessages.empty,
+        },
+        {
+          field: 'address',
+          errorMessage:
+            PageText.Contact.inputFields.address.errorMessages.empty,
+        },
+        {
+          field: 'postalCode',
+          errorMessage:
+            PageText.Contact.inputFields.postalCode.errorMessages.empty,
+        },
+        {
+          field: 'city',
+          errorMessage: PageText.Contact.inputFields.city.errorMessages.empty,
+        },
+        {
+          field: 'phoneNumber',
+          errorMessage:
+            PageText.Contact.inputFields.phoneNumber.errorMessages.empty,
+        },
+      );
+
+      return fields;
+  }
+};
+
+export const formInputValidator = (context: any) => {
+  const inputErrorMessages: InputErrorMessages = {};
+
+  let fieldsToValidate;
+
+  if (context.stateSubmitted) {
```

Comment:
A bit hard to understand what "stateSubmitted" is. Also, when would we have a case where "stateSubmitted" is not set and we would want to validate?

Reply:
Will update the variable to be more specific.